ykrt/src/compile/jitc_yk/opt/mod.rs

```diff
@@ -778,9 +785,24 @@ impl Opt {
         iidx: InstIdx,
         mut pa_inst: PtrAddInst,
     ) -> Result<(), CompilationError> {
-        let mut off = 0;
+        // Although we store ptr_add offsets as an i32, LLVM semantics require the offset to wrap
+        // as though they were "pointer index typed" (a pointer-sized integer, for addrspace 0, the
+        // only address space we support right now).
+        let mut off: isize = 0;
+
         loop {
-            off += pa_inst.off();
+            // LLVM semantics permit silent wrapping.
+            // The unwrap() can't fail on platforms we care about.
+            off = match off.checked_add(isize::try_from(pa_inst.off()).unwrap()) {
```

Comment:
This would be clearer as just `off.checked_add(...).unwrap()` without the `match` IMHO. The comment is fine *but* I wouldn't mind if it was a couple of lines shorter.

[You'll notice that I'm very protective of the optimiser: it's so dangerous if we get things wrong that code quality *and* code size are constraints here that we don't face elsewhere.]

Reply:
Did you mean `wrapping_add`?

Reply:
No. What I meant is : the `match` is just a bloated way of saying "we can't handle the None case", which is what `unwrap` is meant to do.